src/ds/redblacktree.h

```diff
@@ -106,20 +147,26 @@ namespace snmalloc
       {
         return ptr != t.ptr;
       }
+      ///@}
 
-      H* addr()
+      bool is_null()
       {
-        return ptr;
+        return Rep::get(ptr) == Rep::null;
       }
 
-      bool is_null()
+      /**
+       * Return the reference in some printable format defined by the
+       * representation.
+       */
+      auto printable()
       {
-        return Rep::get(ptr) == Rep::null;
+        return Rep::printable(ptr);
       }
     };
 
     // Root field of the tree
-    H root{};
+    K root_val{Rep::root};
+    H root{&root_val};
```

Comment:
It needs to be `{H{&root_val}}`, but yes, you're right.  I was confused by the equality comparison over references and assumed that the root node reference needed to be stable, rather than the storage for the root node.  Fixed.